index.d.ts

```diff
@@ -1123,7 +1123,7 @@ declare module 'mongoose' {
     pathType(path: string): string;
 
     /** Registers a plugin for this schema. */
-    plugin(fn: (schema: Schema, opts?: any) => void, opts?: any): this;
+    plugin<T extends Document = DocType, L extends Model<T> = M>(fn: (schema: Schema<T, L, SchemaDefinitionType>, opts?: any) => void, opts?: any): this;
```

Comment:
Why are the generics necessary? Adding `schema: Schema<DocType, Model<DocType>, SchemaDefinitionType>` should be sufficient for this case, shouldn't it?

Reply:
That's correct, it would be sufficient for the case above. I was following the same pattern as found with post/pre methods just below. I can update according to your suggestion if you would prefer that?

Reply:
Yes please, I'd rather avoid adding unnecessary generic params if possible.